docs/REQUIRES_VS_IMPORTS.md

```diff
@@ -0,0 +1,470 @@
+  name: 'MenuView',
+  extends: 'foam.core.menu.VerticalMenu',
+
+  imports: [
+    'currentProgram',    // Live program object
+    'isMenuOpen'        // Boolean state value
+  ]
+  // Access: this.currentProgram (the actual object)
+  //         this.isMenuOpen (the actual boolean value)
+});
+```
+
+**Service Injection:**
+```javascript
+// File: foam3/src/foam/u2/wizard/debug/WizardInspector.js
+foam.CLASS({
+  package: 'foam.u2.wizard.debug',
+  name: 'WizardInspector',
+  extends: 'foam.u2.Controller',
+
+  imports: [
+    'wizardController',    // Live controller instance
+    'crunchController'     // Live controller instance
+  ]
+  // These are actual running controller objects, not classes
+});
+```
+
+### When to Use `imports`
+- ✅ When you need **services** or **data** from the application context
+- ✅ For **dependency injection** of runtime values
+- ✅ When accessing **DAOs**, **services**, or **configuration** from parent contexts
+- ✅ For **logging services**, **authentication context**, or **global state**
+- ✅ When you want **loose coupling** and **testability**
+- ✅ When the dependency might not always be available (use `?`)
+
+## `exports:` - Making Services Available
+
+### Purpose
+The flip side of `imports:` - declares what objects or values you want to make available to descendants in the context chain.
+
+### Syntax
+```javascript
+foam.CLASS({
+  name: 'MyService',
+  
+  exports: [
+    'userDAO',           // Export userDAO property to children
+    'this as myService', // Export entire instance
+    'log as logger'      // Export with alias
+  ]
+});
+```
+
+### How It Works
+- Exported values become available in the context for child components
+- Child components can `import` these exported values
+- Creates a context chain where services flow down to descendants
+- Enables **inversion of control** architecture
+
+## Key Differences Summary
+
+| Aspect | `requires` | `imports` |
+|--------|------------|-----------|
+| **Purpose** | Class name resolution | Service injection |
+| **Result** | Factory methods | Direct property access |
+| **Type** | Classes/constructors | Instances/values/services |
+| **Source** | Static class definitions | Runtime execution context |
+| **Analogy** | `import` in Java, `require()` in JS | Dependency injection frameworks |
+| **Creation** | `this.ClassName.create()` | `this.serviceName` |
+| **Coupling** | Compile-time dependencies | Runtime loose coupling |
+| **Testability** | Hard to mock classes | Easy to inject test doubles |
+| **Network** | Local class resolution only | Can import across network |
+| **Optional** | No | Yes (with `?`) |
+
+## Dependency Injection Benefits
+
+### Without Dependency Injection (Tight Coupling)
+```javascript
+// BAD: Going fishing for dependencies
+foam.CLASS({
+  name: 'BadExample',
+  methods: [
+    function init() {
+      // Tight coupling - class knows where to find dependencies
+      this.userDAO = this.__context__.lookup('userDAO');
```

Comment:
The context is still loose-coupling, it's just a little more work. Tight-coupling would be doing something like this.frame.controller.session.daoAccessor.userDAO, which requires you to know the architecture of your application and follow it explicitly to find services you need. Then if the structure every changed, all of your client code would break.